### ISSUE TITLE
[Idea] Automatically spawn into advanced classes based on momentum

### ISSUE THREAD

**Issue by DolceTriade** [Unvanquished/gameplay#38]
As the game goes on, weaker aliens (mara/mantis/etc) get far weaker and of litttle use. Instead of having advanced versions of base classes for an additional evos, we should just unlock abilities for base classes based on momentum. This will allow for aliens to have more base destructive AoE attacks as the game progresses.

**Comment by slipher** [comment]
I like it.

Should you lose advanced abilities when momentum goes down? Or do you keep the advanced form based on whether it was available when you evolved? I'd prefer the latter because it would be too frustrating if you lose an advanced ability 1 minute after getting it (without even dying).

**Comment by DolceTriade** [comment]
Yeah, you keep the functionality you have when you evolved. Consequently, that means that you do not get newly unlocked abilities until you evolve. This can be used in conjunction with the idea that you can de-evolve near the OM (I can't seem to find this issue though...)

**Comment by slipher** [comment]
Devolving is already implemented.

**Comment by illwieckz** [comment]
Outside of liking or not the idea, we are not ready to enable it with the current implementation of the game as it would brings break all games with bots (and maybe some games without bots) as players would spawn with large classes on areas they can't leave with such classes (even if they can spawn with).

If I'm right we delayed the evolution of bots so they can leave the spawn place as dretch before evolving, this to workaround multiples bugs like bots spawning directly as dragoon (evolving immediately on spawn) and being stuck in disconnected navmesh in Perseus. I'm not finding anymore the related thread though, but this is a very problematic issue that comes with spawning as large classes.

**Comment by illwieckz** [comment]
One variant may be that once momentum is unlocked, some advanced classes may be cheaper or free. With the ability to preselect which class to prefer on every spawn, this may provide a similar gameplay while preventing the large-class-spawning issue in problematic maps (player would just not preselect huge class on purpose).

**Comment by slipher** [comment]
@illwieckz The idea is not to spawn as a large form rather than granger/dretch. The idea is that instead of "Marauder" and "Advanced Marauder" in the evolve menu, there is only Marauder, which automatically acquires advanced abilities depending on the momentum.

**Comment by bmorel** [comment]
Yes, the delay before evolving is a workaround by @illwieckz to prevent bots to be stuck in many places. Another place to mention is uss tremor's areas above several ladders.
I [had to edit](https://github.com/bmorel/unvanquished-layouts/commit/190976b6e6d1295c7c831b3c2125abc57ec79417) a PvE layout to remove an egg for that very reason, because the delay is still too small (but increasing the delay would a bad idea anyway).

On that point (AIs) the ideal solution would be to evolve when the bot really thinks it's going to fight, but... well, I don't really see how to implement this.

In any case, I don't think AIs shuold be considered too much when doing gameplay changes. Bots are important, sure, because without them the game would just loose it's remaining players, *but* bots can be fixed and improved, and you can see it if you play "nightly build server" and then "0.52" game.
Those bots are even still behind my own changes for various features (crouching and lesser auto-jump are probably the most spectacular changes I have).
All is required to get better bots is someone to spend time improving them, instead of hiding behind the "bots are a substitute for real players and can't play this game because it's too complicated" excuse.

@slipher mentions that this would be weird to loose functions when momentum decrease: well, this is an annoying feature in current gameplay, that I would not be against removing. Anyway, it has to be noted that if a player is an adv granger and momentum decrease (for example because said user is moving base to ceilings, it happens *very* often) the player does not loose his abilities, so the proposed system should probably just work like that.

As for the proposal itself, I'm not convinced it's a good idea. Nor that it's a bad idea. Basically: I don't know what to think about it.

Lastly, I've seen @necessarily-equal saying on IRC that currently aliens receive all the buffs, and indeed they do. And they're the most powerful faction of the game in expert hands.
Buffing them even more might be a very bad idea for balance.
